### PR TITLE
Improve conversion of reactive handlers

### PR DESCRIPTION
**Pull request by tsegismont**
See #314

### CODE REVIEW

**Comment by tsegismont** [comment]
@vietj when this is merged, I'll backport to `4.x`

@jponge can you take care of porting this to Mutiny Vert.x bindings?

File: rx-java2-gen/src/main/java/io/vertx/lang/reactivex/Helper.java [hunk 1]
```diff
@@ -22,4 +44,16 @@ public static Class unwrap(Class<?> type) {
...
   }
+
+  /**
+   * Convert a handler for a generated reactive type to a handler for the corresponding core type.
+   */
+  public static <CORE, REACTIVE> Handler<CORE> convertHandler(Handler<REACTIVE> rxHandler, Function<CORE, REACTIVE> mapper) {
+    RxGen annotation = rxHandler.getClass().getAnnotation(RxGen.class);
+    if (annotation != null && rxHandler instanceof RxDelegate) {
+      RxDelegate rxDelegate = (RxDelegate) rxHandler;
+      return (Handler<CORE>) annotation.value().cast(rxDelegate.getDelegate());
```
Comment (tsegismont) [inline rx-java2-gen/src/main/java/io/vertx/lang/reactivex/Helper.java:55]: @vietj I'm not sure there is much interest in `annotation.value().cast()` here...

Maybe we could get away with `return (Handler<CORE>) rxDelegate.getDelegate();`

File: rx-java3-gen/src/main/java/io/vertx/lang/rxjava3/Helper.java [hunk 2]
```diff
@@ -22,4 +44,15 @@ public static Class unwrap(Class<?> type) {
     }
     return type;
   }
+
+  /**
+   * Convert a handler for a generated reactive type to a handler for the corresponding core type.
+   */
+  public static <CORE, REACTIVE> Handler<CORE> convertHandler(Handler<REACTIVE> rxHandler, Function<CORE, REACTIVE> mapper) {
+    if (rxHandler.getClass().isAnnotationPresent(RxGen.class) && rxHandler instanceof RxDelegate) {
```
Comment (vietj) [inline rx-java3-gen/src/main/java/io/vertx/lang/rxjava3/Helper.java:52]: I believe we don't need to check annotation presence here

**Comment by jponge** [comment]
I don't have the bandwidth until January, if you want to offer a backport no problem

**Comment by tsegismont** [comment]
> I don't have the bandwidth until January, if you want to offer a backport no problem

Same here. It would be nice to have it in the next bugfix version of the Mutiny bindings that will go into Quarkus. But I'm not sure when this will happen. Not likely before January. So let's talk about this again next year :wink:

**Review by vietj — APPROVED** [review]

**Pull request merged**